open-api/rest-catalog-open-api.yaml

```diff
@@ -2068,6 +2165,145 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        A flexible JSON object that contains information provided by the server,
+        to be utilized by clients for distributed planning. However it is recommended that servers
+        provide a plan-task-id, as well as a location to a manifest.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - data-file
+      properties:
+        data-file:
+          $ref: '#/components/schemas/ContentFile'
+        partition:
+          type: object
+          additionalProperties:
+            type: string
+        size-bytes:
+          type: number
+        start:
+          type: number
+        length:
+          type: number
+        estimated-rows-count:
+          type: number
+        delete-files:
+          type: array
+          items:
+            $ref: '#/components/schemas/ContentFile'
+        schema:
+          $ref: '#/components/schemas/Schema'
+        spec:
+          $ref: '#/components/schemas/PartitionSpec'
+        residual-filter:
+          $ref: '#/components/schemas/Expression'
+
+    ContentFile:
+      type: object
+      required:
+        - file-path
```

Comment:
similar comment, please check ContentFileParser.fromJson for what are required

Reply:
Will rebase changes on Drew pr